kratos/conditions/penalty_coupling_condition.cpp

```diff
@@ -0,0 +1,221 @@
+//    |  /           |
+//    ' /   __| _` | __|  _ \   __|
+//    . \  |   (   | |   (   |\__ `
+//   _|\_\_|  \__,_|\__|\___/ ____/
+//                   Multi-Physics
+//
+//  License:         BSD License
+//                   Kratos default license: kratos/license.txt
+//
+//  Main authors:    Tobias Tescheamacher
+//
+
+// System includes
+
+// External includes
+#include "conditions/penalty_coupling_condition.h"
+
+// Project includes
+
+namespace Kratos
+{
+
+    void PenaltyCouplingCondition::CalculateAll(
+        MatrixType& rLeftHandSideMatrix,
+        VectorType& rRightHandSideVector,
+        ProcessInfo& rCurrentProcessInfo,
+        const bool CalculateStiffnessMatrixFlag,
+        const bool CalculateResidualVectorFlag)
+    {
+        KRATOS_TRY
+
+        DofVariablesContainer const& r_dof_variables = this->GetProperties().GetValue(DOF_VARIABLES);
```

Comment:
:+1:

Reply:
Maybe not the most descriptive name. BTW i prefer the const at the begining

Reply:
let's see as soon as I have added the process, I guess it will get clearer, also the naming.